src/Tasks.UnitTests/MSBuild_Tests.cs

```diff
@@ -1408,5 +1407,76 @@ public void NonexistentTargetFailsAfterSkipped()
             // 2nd invocation of t_missing should fail the build resulting in target not found error (MSB4057)
             logger.FullLog.ShouldContain("MSB4057");
         }
+
+        [Fact]
+        public void MSBuildTaskPassesTaskIdToSpawnedBuilds()
+        {
+            string projectFile1 = ObjectModelHelpers.CreateTempFileOnDisk(@"
+                <Project>
+                    <Target Name=`Build`>
+                        <Message Text=`test`/>
+                    </Target>
+                </Project>");
+
+            string projectFile2 = ObjectModelHelpers.CreateTempFileOnDisk(@"
+                <Project>
+                    <Target Name=`Build`>
+                        <MSBuild Projects=`" + projectFile1 + @"` Targets=`Build` />
+                    </Target>	
+                </Project>");
+
+            try
+            {
+                Project project = new Project(projectFile2);
+                var logger = new MSBuildTaskIdLogger();
+
+                Assert.True(project.Build(logger));
+
+                var text = logger.Text;
+                var taskId = Regex.Match(text, @"TaskId=(\d+)").Groups[1].Value;
```

Comment:
You can use [MockLogger.ProjectStartedEvents](https://github.com/cdmihai/msbuild/blob/7bcb2f961093e87277439420d351a9f3f3a8cdb8/src/Shared/UnitTests/MockLogger.cs#L87) to avoid string matching

Reply:
Good idea, fixed!